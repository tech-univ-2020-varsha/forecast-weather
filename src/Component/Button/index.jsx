import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';

const Button = ({ testId, children }) => (
  <button data-testid={testId} className={styles.getWeatherBtn} type="submit">
    <div className={styles.btnText}>{children}</div>
  </button>
);

export default Button;


Button.propTypes = {
  testId: propTypes.string.isRequired,
};
