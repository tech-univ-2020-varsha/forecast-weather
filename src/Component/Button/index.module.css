.getWeatherBtn{
    background-color: bisque;
    text-align: center;
   border-radius: 1em;
   background-color: rgb(234, 244, 246);
   border: solid gray; ;
}
.btnText{
    padding: 1em;
    color: black;
    font-family: Arial, Helvetica, sans-serif;
    font-weight: bold;

}