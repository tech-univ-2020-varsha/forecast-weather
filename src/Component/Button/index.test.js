import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import Button from './index';

describe('the Button component', () => {
  it('should render button correctly', () => {
    const { asFragment } = render(<Button testId="test-btn">Test text</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
