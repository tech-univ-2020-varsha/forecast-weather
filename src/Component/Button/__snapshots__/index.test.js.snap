// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Button component should render button correctly 1`] = `
<DocumentFragment>
  <button
    data-testid="test-btn"
    type="submit"
  >
    <div>
      Test text
    </div>
  </button>
</DocumentFragment>
`;
